lib/widgets/message_list.dart

```diff
@@ -97,6 +97,31 @@ class MessageListAppBarTitle extends StatelessWidget {
   }
 }
 
+class ScrollToBottomButton extends StatelessWidget {
+  const ScrollToBottomButton({super.key, required this.scrollController, required this.visibleValue});
+  final ValueNotifier<bool> visibleValue;
+  final ScrollController scrollController;
+
+  Future<void> _navigateToBottom(BuildContext context) async {
+    scrollController.animateTo(0, duration: const Duration(milliseconds: 300), curve: Curves.easeIn);
```

Comment:
This feels pretty abrupt when the user is scrolled far up — and it gets a bit choppy even in a release build. It'll presumably get more so when longer history becomes possible with #78 (coming soon). How about let's try putting a maximum scrolling speed on it?

For a unit of scale, 8000 (logical) pixels per second is the maximum scrolling speed that Flutter, or Android native, or I believe also iOS native, will clamp to when the user fling-scrolls quickly. (And around 13000 px/sec is the fastest I can physically scroll with my finger, from when I was doing a lot of testing of scroll behavior a few months ago.) So a reasonable max speed could be 8000 px/sec, or some multiplier times that.

Reply:
I've implemented the speed limit in 863724ad52da1c54380c031fb5258aca74be3396 by dynamically setting the duration based on distance.